engine/src/execution_graph/logic_controllers/CacheMachine.cpp

```diff
@@ -319,6 +355,20 @@ ConcatenatingCacheMachine::ConcatenatingCacheMachine(size_t bytes_max_size)
 {
 }
 
+bool checkStrings(std::vector<std::unique_ptr<ral::frame::BlazingTable>> & holder_samples, CacheData & cache_data){
+	if(holder_samples.size()>=1){
+		for(int i=0;i<holder_samples[0]->view().num_columns();i++){
```

Comment:
Add some checks here to make sure that all tables have the same number of columns. THat way in case that is not the case we can raise an error and not get a segfault.

Also, please rename this function. `checkStrings` is not very informative. It should be more like `checkIfConcatenatingStringsWillOverflowStringLength` or `checkIfCumulativeStringSizeIsTooBig` and please add an explanation for what the function does and what it should return

Reply:
I renamed the function and added some checks.